samza-core/src/main/java/org/apache/samza/operators/impl/WindowOperatorImpl.java

```diff
@@ -268,100 +267,40 @@ private void cancelTrigger(TriggerKey<WK> triggerKey, boolean shouldRemove) {
   }
 
   /**
-   * Implementation of the {@link TriggerContext} that allows {@link TriggerImpl}s to schedule and cancel callbacks.
-   */
-  public class TriggerContextImpl implements TriggerContext {
-    private final TriggerKey<WK> triggerKey;
-
-    public TriggerContextImpl(TriggerKey<WK> triggerKey) {
-      this.triggerKey = triggerKey;
-    }
-
-    @Override
-    public Cancellable scheduleCallback(Runnable runnable, long callbackTimeMs) {
-      TriggerCallbackState timerState = new TriggerCallbackState(triggerKey, runnable, callbackTimeMs);
-      pendingCallbacks.add(timerState);
-      LOG.trace("Scheduled a new callback: {} at {} for triggerKey {}", new Object[] {runnable, callbackTimeMs, triggerKey});
-      return timerState;
-    }
-  }
-
-  /**
-   * State corresponding to pending timer callbacks scheduled by various {@link TriggerImpl}s.
-   */
-  private class TriggerCallbackState implements Comparable<TriggerCallbackState>, Cancellable {
-
-    private final TriggerKey<WK> triggerKey;
-    private final Runnable callback;
-
-    // the time at which the callback should trigger
-    private final long scheduledTimeMs;
-
-    private TriggerCallbackState(TriggerKey<WK> triggerKey, Runnable callback, long scheduledTimeMs) {
-      this.triggerKey = triggerKey;
-      this.callback = callback;
-      this.scheduledTimeMs = scheduledTimeMs;
-    }
-
-    private Runnable getCallback() {
-      return callback;
-    }
-
-    private long getScheduledTimeMs() {
-      return scheduledTimeMs;
-    }
-
-    @Override
-    public int compareTo(TriggerCallbackState other) {
-      return Long.compare(this.scheduledTimeMs, other.scheduledTimeMs);
-    }
-
-    @Override
-    public boolean cancel() {
-      LOG.trace("Cancelled a callback: {} at {} for triggerKey {}", new Object[] {callback, scheduledTimeMs, triggerKey});
-      return pendingCallbacks.remove(this);
-    }
-  }
-
-  /**
    * State corresponding to a created {@link TriggerImpl} instance.
```

Comment:
Now, with the cleanup of code, this class seems to be implementing "handler" functions of a TriggerImpl. Maybe better to name as TriggerImplHandler?